native/acter/src/api/stream.rs

```diff
@@ -1134,73 +436,119 @@ impl TimelineStream {
                     bail!("No permission to send message in this room");
                 }
 
-                let timeline_event = room.event(&event_id).await?;
-                let event_content = timeline_event
-                    .event
-                    .deserialize_as::<RoomMessageEvent>()
-                    .context("Couldn't deserialise event")?;
-
-                let mut sent_by_me = false;
-                if let Some(user_id) = client.user_id() {
-                    if user_id == event_content.sender() {
-                        sent_by_me = true;
+                let content = match *draft {
+                    MsgContentDraft::TextPlain { body } => {
+                        RoomMessageEventContentWithoutRelation::text_plain(body)
                     }
-                }
-                if !sent_by_me {
-                    bail!("Can't edit an event not sent by own user");
-                }
-
-                let location_content = LocationMessageEventContent::new(body, geo_uri);
-                let mut edited_content =
-                    RoomMessageEventContent::new(MessageType::Location(location_content.clone()));
-                let replacement = Replacement::new(
-                    event_id.to_owned(),
-                    MessageType::Location(location_content).into(),
-                );
-                edited_content.relates_to = Some(Relation::Replacement(replacement));
-
-                timeline.send(edited_content.into()).await;
-                Ok(true)
-            })
-            .await?
-    }
-
-    pub async fn send_location_reply(
-        &self,
-        body: String,
-        geo_uri: String,
-        event_id: String,
-    ) -> Result<bool> {
-        if !self.is_joined() {
-            bail!("Can't send reply as location to a room we are not in");
-        }
-        let room = self.room.clone();
-        let my_id = room
-            .client()
-            .user_id()
-            .context("User not found")?
-            .to_owned();
-        let timeline = self.timeline.clone();
-        let event_id = EventId::parse(event_id)?;
-
-        RUNTIME
-            .spawn(async move {
-                let member = room
-                    .get_member(&my_id)
-                    .await?
-                    .context("Couldn't find me among room members")?;
-                if !member.can_send_message(MessageLikeEventType::RoomMessage) {
-                    bail!("No permission to send message in this room");
-                }
+                    MsgContentDraft::TextMarkdown { body } => {
+                        RoomMessageEventContentWithoutRelation::text_markdown(body)
+                    }
+                    MsgContentDraft::Image { body, source, info } => {
+                        let info = info.expect("image info needed");
```

Comment:
similar as above with the edit-content...

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1175/commits/d19f8cc4609bc80a191fcfc7e52a58035f83c76f